app/src/main/java/com/codesoom/assignment/TodoHttpHandler.java

```diff
@@ -0,0 +1,130 @@
+package com.codesoom.assignment;
+
+import com.codesoom.assignment.controller.TodoHttpController;
+import com.codesoom.assignment.models.Todo;
+import com.codesoom.assignment.validate.TodoValidator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+
+import java.io.*;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public class TodoHttpHandler implements HttpHandler {
+    private final TodoHttpController todoHttpController;
+    private final TodoValidator todoValidator;
+    public TodoHttpHandler() {
+        this.todoHttpController = new TodoHttpController();
+        this.todoValidator = new TodoValidator();
+    }
+
+    @Override
+    public void handle(HttpExchange exchange) {
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            String requestMethod = exchange.getRequestMethod();
+            String path = exchange.getRequestURI().getPath().substring(1);
+            OutputStream outputStream = new ByteArrayOutputStream();
+            InputStream requestBody = exchange.getRequestBody();
+            OutputStream responseBody = exchange.getResponseBody();
+
+            try {
+                todoValidator.isContainsInvalidId(path);
+            } catch (IllegalArgumentException e) {
+                objectMapper.writeValue(outputStream, Arrays.asList());
+                exchange.sendResponseHeaders(404, outputStream.toString().getBytes().length);
+                e.printStackTrace();
+            }
+
+            if (requestMethod.equals("GET") &&
```

Comment:
이렇게 하면 만약 `requestMethod`가 `null`이어도 NPE가 던져지지 않게 됩니다. 많이 쓰이는 기법이니 기억해두세요.
```suggestion
            if ("GET".equals(requestMethod) &&
```

Reply:
@johngrib 안녕하세요. 저는 NPE가 던져져야 하는 곳에서 던져지지 않는다면, 에러가 아니라 버그로 연결될 수 있어 "".equals를 지양하는편인데, 그래도 바꾸는게 좋을까요?

Reply:
네 이런 상황에서 requestMethod의 null을 검사하는 것이 목적이라면 이 if 문에 들어서기 전에 의도적으로 예외를 던지는 것이 낫습니다. 말씀하신대로 NPE를 던지는 것을 목표로 두고 이 상태로 놔두게 되면 이 코드의 의도가 한 가지를 넘게 되죠.